packages/contracts-bedrock/test/L1/OPContractsManager.t.sol

```diff
@@ -397,6 +407,18 @@ contract OPContractsManager_Upgrade_Test is OPContractsManager_Upgrade_Harness {
         // Run the upgrade test and checks
         runUpgradeTestAndChecks(_nonUpgradeController);
     }
+
+    function test_upgrade_duplicateL2ChainId_succeeds() public {
+        // Deploy a new OPChain with the same L2 chain ID as the current OPChain
+        Deploy deploy = Deploy(address(uint160(uint256(keccak256(abi.encode("optimism.deploy"))))));
+        IOPContractsManager.DeployInput memory deployInput = deploy.getDeployInput();
+        deployInput.l2ChainId = l2ChainId;
+        deployInput.saltMixer = "v2.0.0";
```

Comment:
this saltMixer is imperfect because it doesn't try to perfectly replicate what it would be in `opcm.upgrade()`, but even if we predict the SystemConfig address it clearly won't be the same as an already existing SystemConfig.